### PR TITLE
This upgrades our gym dependency

### PR DESCRIPTION
**Pull request by cash**
* We now need to install classic_control to get cartpole. 
 * moved gym minigrid to the standard install. 
 * added a check for whether atari is installed.
 * updated unit tests with rewards checks since gym 0.22.0 changes the RNG used

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.9]
```
Comment (coreylowman) [inline .github/workflows/pytest.yml:9]: Is this because new gym requires 3.9?

**Review by coreylowman — APPROVED** [review]

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (cash) [reply .github/workflows/pytest.yml:9]: No, this was me trying to figure out why all the unit tests were failing in 3.7 and 3.8. I will restore that now that I've figured it out.

**Comment by cash** [comment]
This still uses the seed() function in gym which is deprecated. I was running into problems updating that and felt it was better to get this working first.

**Comment by cash** [comment]
@coreylowman should I switch the gym dependency to 0.22.0 or greater? That's when all these changes occurred that we now require like the termination observation.

**Comment by coreylowman** [comment]
@cash sure 0.22.0 sounds good. Are we doing a 2.0.0 release after this is merged?

**Comment by cash** [comment]
I don't know of anything else that needs to be in a 2.0.0 release. @ginoperrotta anything?

**Pull request merged**